odxtools/isotp_state_machine.py

```diff
@@ -28,29 +29,29 @@ class IsoTpStateMachine:
         "([a-zA-Z0-9_-]*) *([0-9A-Fa-f ]*) *\[[0-9]*\] *([ 0-9A-Fa-f]*)")
     can_log_frame_re = re.compile("\([0-9.]*\) *([a-zA-Z0-9_-]*) ([0-9A-Fa-f]*)#([0-9A-Fa-f]*)")
 
-    def __init__(self, can_rx_ids):
+    def __init__(self, can_rx_ids: Union[int, List[int]]):
         if isinstance(can_rx_ids, int):
             can_rx_ids = [can_rx_ids]
 
         self._can_rx_ids = can_rx_ids
         assert isinstance(self._can_rx_ids, list)
 
         self._telegram_specified_len = [0] * len(can_rx_ids)
-        self._telegram_data = [None] * len(can_rx_ids)
+        self._telegram_data: List[Optional[bytearray]] = [None] * len(can_rx_ids)
         self._telegram_last_rx_fragment_idx = [0] * len(can_rx_ids)
 
-    def decode_rx_frame(self, rx_id, data, do_yield=False):
+    def decode_rx_frame(self, rx_id: int, data: bytes) -> Iterable[Tuple[int, bytes]]:
         """Handle the ISO-TP state transitions caused by a CAN frame.
 
-        E.g., add some data to a telegram, etc.
+        E.g., add some data to a telegram, etc. Returns a generator of
+        (receive_id, payload_data) tuples.
         """
         try:
             telegram_idx = self._can_rx_ids.index(rx_id)
         except ValueError:
-            return False  # unknown CAN ID
+            return  # unknown CAN ID
```

Comment:
behavior change?

Reply:
not really: this method was always called with `do_yield = True`, and I'm honestly not sure anymore why I've implemented that option in the first place...